mobile-app/app/screens/TransactionAuthorization/PasscodePrompt.tsx

```diff
@@ -36,148 +40,140 @@ interface PasscodePromptProps {
-                    {props.additionalMessageUrl}
-                  </ThemedText>
-                </View>
-              )}
+            }
 
-              <View style={tailwind('px-8 text-sm text-center mb-6')}>
-                <ThemedText
+            {// show loading message
+              props.status === TransactionStatus.SIGNING &&
+            (
+              <ThemedTextV2
+                testID='txn_authorization_message'
+                dark={tailwind('text-mono-dark-v2-700')}
+                light={tailwind('text-mono-light-v2-700')}
+                style={tailwind('px-8 text-sm text-center font-normal-v2')}
+              >
+                {translate('screens/UnlockWallet', props.loadingMessage)}
+              </ThemedTextV2>
+            )
+            }
+            {// show default message
+              (!props.isRetry && props.status !== TransactionStatus.AUTHORIZED && props.status !== TransactionStatus.SIGNING) &&
+                <ThemedTextV2
                   testID='txn_authorization_message'
-                  dark={tailwind('text-gray-400')}
-                  light={tailwind('text-gray-500')}
-                  style={tailwind('p-2 px-8 text-sm text-center mb-2')}
+                  dark={tailwind('text-mono-dark-v2-700')}
+                  light={tailwind('text-mono-light-v2-700')}
+                  style={tailwind('px-8 text-sm text-center font-normal-v2')}
                 >
-                  {props.message}
-                </ThemedText>
+                  {translate('screens/UnlockWallet', props.message)}
+                </ThemedTextV2>
 
-                {
-                  props.transaction?.description !== undefined && (
-                    <ThemedText
-                      testID='txn_authorization_description'
-                      dark={tailwind('text-gray-400')}
-                      light={tailwind('text-gray-500')}
-                      style={tailwind('text-sm text-center')}
-                    >
-                      {props.transaction.description}
-                    </ThemedText>
-                  )
+            }
+            {// hide description when passcode is incorrect or verified.
+            (props.transaction?.description !== undefined && !props.isRetry && props.status !== TransactionStatus.AUTHORIZED) &&
+              (
+                <ThemedTextV2
+                  testID='txn_authorization_description'
+                  dark={tailwind('text-mono-dark-v2-700')}
+                  light={tailwind('text-mono-light-v2-700')}
+                  style={tailwind('text-sm text-center font-normal-v2')}
+                >
+                  {props.transaction.description}
+                </ThemedTextV2>
+              )
+            }
+            {// upon retry: show remaining attempt allowed
+                  (props.isRetry && props.attemptsRemaining !== undefined && props.attemptsRemaining !== props.maxPasscodeAttempt && props.status !== TransactionStatus.SIGNING)
```

Comment:
no need of turnery condition here

Reply:
> 

If we follow the figma exactly there will be an issue with the text unless we give the container a set width. In my opinion that's not ideal

Reply:
use `&&` operator in place of `<condition> ? <code> : null`

Reply:
ah i got the wrong context hahahhaaha